packages/flutter/lib/src/widgets/animated_grid.dart

```diff
@@ -0,0 +1,714 @@
+  _ActiveItem.index(this.itemIndex)
+      : controller = null,
+        removedItemBuilder = null;
+
+  final AnimationController? controller;
+  final AnimatedGridRemovedItemBuilder? removedItemBuilder;
+  int itemIndex;
+
+  @override
+  int compareTo(_ActiveItem other) => itemIndex - other.itemIndex;
+}
+
+/// A scrolling container that animates items when they are inserted or removed
+/// in a grid.
+///
+/// This widget's [AnimatedGridState] can be used to dynamically insert or
+/// remove items. To refer to the [AnimatedGridState] either provide a
+/// [GlobalKey] or use the static [of] method from an item's input callback.
+///
+/// This widget is similar to one created by [GridView.builder].
+///
+/// {@tool dartpad}
+/// This sample application uses an [AnimatedGrid] to create an effect when
+/// items are removed or added to the grid.
+///
+/// ** See code in examples/api/lib/widgets/animated_grid/animated_grid.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+/// * [SliverAnimatedGrid], a sliver which animates items when they are inserted
+///   or removed from a grid.
+/// * [SliverAnimatedList], a sliver which animates items added and removed from
+///   a list instead of a grid.
+/// * [AnimatedList], which animates items added and removed from a list instead
+///   of a grid.
+class AnimatedGrid extends StatefulWidget {
+  /// Creates a scrolling container that animates items when they are inserted
+  /// or removed.
+  const AnimatedGrid({
+    super.key,
+    required this.itemBuilder,
+    required this.gridDelegate,
+    this.initialItemCount = 0,
+    this.scrollDirection = Axis.vertical,
+    this.reverse = false,
+    this.controller,
+    this.primary,
+    this.physics,
+    this.shrinkWrap = false,
+    this.padding,
+    this.clipBehavior = Clip.hardEdge,
+  })  : assert(itemBuilder != null),
+        assert(initialItemCount != null && initialItemCount >= 0);
+
+  /// Called, as needed, to build grid item widgets.
+  ///
+  /// Grid items are only built when they're scrolled into view.
+  ///
+  /// The [AnimatedGridItemBuilder] index parameter indicates the item's
+  /// position in the grid. The value of the index parameter will be between 0
+  /// and [initialItemCount] plus the total number of items that have been
+  /// inserted with [AnimatedGridState.insertItem] and less the total number of
+  /// items that have been removed with [AnimatedGridState.removeItem].
+  ///
+  /// Implementations of this callback should assume that
+  /// [AnimatedGridState.removeItem] removes an item immediately.
+  final AnimatedGridItemBuilder itemBuilder;
+
+  /// A delegate that controls the layout of the children within the
+  /// [AnimatedGrid].
+  ///
+  /// See also:
+  ///
+  ///  * [SliverGridDelegateWithFixedCrossAxisCount], which creates a layout with
+  ///    a fixed number of tiles in the cross axis.
+  ///  * [SliverGridDelegateWithMaxCrossAxisExtent], which creates a layout with
```

Comment:
Offline we chatted about adding convenience constructors like `count` and `extent` in order to spin up a delegate for the user. Can you add those?
Reference: https://github.com/flutter/flutter/issues/113107